lib/read.js

```diff
@@ -36,67 +43,56 @@ var Read = Juttle.proc.source.extend({
+        //cannot order by time if no reduce optimization
+        if (this.timeField && this.optimization_info.type !== 'reduce') {
+            this.baseQuery = this.baseQuery.orderBy(this.timeField, this.tailOptimized ? 'desc' : 'asc');
         }
-    },
+    }
+
+    periodicLiveRead() {
+        return !!this.timeField;
+    }
 
-    validateOptions: function(options) {
-        var unknown = _.difference(_.keys(options), this.allowed_options);
+    validateOptions(options) {
+        var unknown = _.difference(_.keys(options), ALLOWED_OPTIONS);
         if (unknown.length > 0) {
-            throw this.compile_error('RT-UNKNOWN-OPTION-ERROR', {
+            throw new errors.runtimeError('RT-UNKNOWN-OPTION-ERROR', {
                 proc: 'read-sql',
                 option: unknown[0]
             });
         }
-
         if (_.has(options, 'table') === _.has(options, 'raw')) {
-            throw this.compile_error('RT-MISSING-OPTION-ERROR', { option: "choose only one: table,raw" });
+            throw new Error("choose only one option: table,raw.");
         }
 
         //XXX also disallow ANY other options with raw
-    },
+    }
 
-    setOptions: function(options) {
+    setOptions(options) {
         this.debugOption = options.debug;
-        this.timeField = options.timeField;
-        this.fetchSize = options.fetchSize || this.FETCH_SIZE_LIMIT;
+        this.fetchSize = options.fetchSize || ReadSql.FETCH_SIZE_LIMIT;
         this.raw = options.raw;
-    },
 
-    addFilters: function(filter_ast) {
+        // set timeField IFF options reference time field or interval
+        this.timeField = options.timeField ||
+            (options.from || options.to || options.last ? 'time' : undefined);
+    }
+
+    addFilters(filter_ast) {
         var FilterSQLCompiler = FilterJSCompiler.extend(SQLFilter);
         var compiler = new FilterSQLCompiler({ baseQuery: this.baseQuery });
         return compiler.compile(filter_ast);
-    },
-
-    handleTimeOptions: function(options) {
-        if (options.timeField || options.from || options.to) {
-            var timeField = this.timeField || 'time';
-            logger.debug('sorting by time on field ', timeField);
-
-            if (options.from) {
-                this.from = options.from;
-                this.baseQuery = this.baseQuery.where(timeField, '>=', new Date(options.from.valueOf()));
-                logger.debug('start time', new Date(options.from.valueOf()).toISOString());
-            }
-            if (options.to) {
-                this.to = options.to;
-                this.baseQuery = this.baseQuery.where(timeField, '<', new Date(options.to.valueOf()));
-                logger.debug('end time', new Date(options.to.valueOf()).toISOString());
-            }
-        }
-    },
+    }
 
-    addOptimizations: function(optimization_info) {
+    addOptimizations(optimization_info) {
         var self = this;
```

Comment:
You can get rid of this if you fat arrowed your functions. And I think fat arrowing functions is something we want to do as a part of ES6 conversions.